app/code/Magento/InventoryCatalog/etc/di.xml

```diff
@@ -44,6 +44,13 @@
         <plugin name="adapt_add_stock_status_to_products" type="Magento\InventoryCatalog\Plugin\CatalogInventory\Helper\Stock\AdaptAddStockStatusToProductsPlugin"/>
         <plugin name="adapt_assign_status_to_product" type="Magento\InventoryCatalog\Plugin\CatalogInventory\Helper\Stock\AdaptAssignStatusToProductPlugin"/>
     </type>
+    <type name="Magento\InventoryApi\Model\SourceValidatorChain">
+        <arguments>
+            <argument name="validators" xsi:type="array">
+                <item name="enabled" xsi:type="object">Magento\InventoryCatalog\Model\Source\Validator\PreventDisableValidator</item>
```

Comment:
'default_source_disable_validator' => DefaultSourceDisableValidator

Reply:
I'll update this